doc/user-manual/language/core-language.lagda.rst

```diff
@@ -8,21 +8,182 @@
 Core language
 *************
 
-.. note::
-   This is a stub
+A program in Agda consists of a number of declarations written in an ``*.agda``
+file. A declaration introduces a new identifier and gives its type and
+definition. It is possible to declare:
+
+* :ref:`datatypes <data-types>`
+* :ref:`function definitions <function-definitions>`
+  (including :ref:`mixfix operators <mixfix-operators>` and
+  :ref:`abstract definitions <abstract-definitions>`)
+* local definitions :ref:`let <let-expressions>` and
+  :ref:`where <where-blocks>`
+* :ref:`record types <record-types>` (including
+  :ref:`coinductive records <copatterns-coinductive-records>`)
+* :ref:`modules <module-basics>`
+* :ref:`postulates <postulates>`
+* :ref:`variables <generalization-of-declared-variables>`
+* :ref:`pragmas <pragmas>`
+* :ref:`program options <command-line-options>`
+* :ref:`pattern-synonyms <pattern-synonyms>`, and
+* :ref:`precedence <precedence>` (fixity).
+
+Declarations have a signature part and a definition part. These can appear
+separately in the program. Names must be declared before they are used, but
+by separating the signature from the definition it is possible to define things
+in :ref:`mutual recursion <mutual-recursion>`.
+
+Grammar
+-------
+
+At its core, Agda is a dependently typed lambda calculus. The grammar of terms
+is as follows:
+
+.. code-block:: text
+
+  a ::= x                       -- variable
+      | λ x → a                 -- abstraction
+      | a a                     -- application
+      | (x : a) → a             -- function space
+      | Set[n]                  -- universe
+      | (a)                     -- grouping
+
+
+Syntax overview
+---------------
+
+The syntax of an Agda program is defined in terms of three key components:
+
+* **Expressions** write function bodies and types.
+* **Declarations** declare types, data-types, postulates, records, functions etc.
+* **Pragmas** define program options.
+
+There are also three main levels of syntax, corresponding to different levels
+of interpretation:
+
+* **Concrete** is the high-level sugared syntax, it representing exactly what
+  the user wrote (Agda.Syntax.Concrete).
+* **Abstract**, before typechecking (Agda.Syntax.Abstract)
+* **Internal**, the full-intepreted core Agda terms, typechecked; roughly
+  corresponding to (Agda.Syntax.Internal).
+
+The full pipeline from an ``*.agda`` file towards an executable is:
+
+.. code-block:: text
+
+  *.agda file
+     ==[ parser (Lexer.x + Parser.y) ]==>
+  Concrete syntax
+     ==[ nicifier (Syntax.Concrete.Definitions) ]==>
+  'Nice' concrete syntax
+     ==[ scope checking (Syntax.Translation.ConcreteToAbstract) ]==>
+  Abstract syntax
+     ==[ type checking (TypeChecking.Rules.*) ]==>
+  Internal syntax
+     ==[ Agda.Compiler.ToTreeless ]==>
+  Treeless syntax
+     ==[ different backends (Compiler.MAlonzo.*, Compiler.JS.*, ...) ]==>
+  Executable
```

Comment:
The backends only produce source code (Haskell, JS, ...).  To create an executable, there might be another stage (e.g. calling the GHC compiler).